publisher-sdk/src/main/java/com/criteo/publisher/context/EmailHasher.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.context
+
+import com.criteo.publisher.annotation.Incubating
+import com.criteo.publisher.annotation.Incubating.CONTEXT
+import java.security.MessageDigest
+import java.util.Locale
+
+object EmailHasher {
+
+  /**
+   * Helper function to hash emails for [UserData.HASHED_EMAIL]
+   *
+   * ## Hashing Format
+   * The hashing should be the users’ email address:
+   * - Encoded in UTF-8
+   * - Trimmed of any white space (eg: “test@criteo.com “ should become “test@criteo.com”)
+   * - Converted to lower case
+   * - Hashed with MD5 & output as ASCII text
+   * - Hashed with SHA256 and output as ASCII text
+   */
+  @Incubating(CONTEXT)
+  @JvmStatic
+  fun hash(email: String): String {
+    return email.trim().toLowerCase(Locale.ROOT).toHash("MD5").toHash("SHA-256")
+  }
+
+  private fun String.toHash(type: String): String {
```

Comment:
`toHash` might as well be a simple private method. Any advantage in defining it as an extension function?

Reply:
The `trim` and `toLowerCase` are already extension function that make the code readable from left to right. Having `toHash` defined as a simple method would make the code readable first from left to right, then from right to left:

```
toHash(toHash(email.trim().toLowerCase(), "MD5", "SHA-256"))
``` 

IMHO, this is not readable and would require to add local variables.

Note that as the extension is private, it is only local to `EmailHasher`.